addons/web_tour/static/src/js/tour_manager.js

```diff
@@ -460,7 +460,19 @@ return core.Class.extend(mixins.EventDispatcherMixin, ServicesMixin, {
                 _.each(this._log, function (log) {
                     console.log(log);
                 });
-                console.log(document.body.parentElement.outerHTML);
+                let documentHTML = document.documentElement.outerHTML;
+                // Replace empty iframe tags with their content
+                const iframeEls = Array.from(document.body.querySelectorAll('iframe.o_iframe'));
+                if (iframeEls.length) {
+                    const iframeRegExp = new RegExp('<iframe.*?class=.*?o_iframe.*?></iframe>', 'g');
```

Comment:
- `<iframe></iframe>Many Stuff<iframe class='o_iframe'></iframe>` -> that whole string is matched by your regex. Using `*?` is a source of error IMO. It is often easier (to me at least) to think in term of "what should I not match?". So something like:
```
<iframe[^>]+class=["'][^"']+ etc
```
But there may be better than that, I don't know but definitely need some more checking :wink: 
- Also use the regex syntax definition, it would make it easier to read IMO: `/yourregex/g` instead of `new RegExp('yourregex', 'g')`. My editor would highlight it properly.
- Use `\b` around your class name, that would ensure not matching an iframe with the class `my_stuff_to_iframe` or  `o_iframe_i_dont_like` :wink: (of course unlikely but just a good practice :wink:)

Reply:
Done :+1: